meson.build

```diff
@@ -0,0 +1,1449 @@
+   add_global_arguments('-DBOOST_STACKTRACE_USE_BACKTRACE', language : 'cpp')
+endif
+boostDependency = dependency('boost',
+                             version : '>=1.79.0',
+                             modules : boostModules,
+                             static : true)
+message('Boost:', boostDependency.name(), 'found =', boostDependency.found(), 'version =', boostDependency.version())
+
+#
+# Extra requirements for Boost Stacktrace
+#
+# Per https://www.boost.org/doc/libs/1_76_0/doc/html/stacktrace/configuration_and_build.html, by default
+# Boost.Stacktrace is a header-only library.  However, you get better results by linking (either statically or
+# dynamically) with a helper library.  Of the various options, it seems like boost_stacktrace_backtrace gives the most
+# functionality over the most platforms.  This has dependencies on:
+#   - libdl on POSIX platforms -- but see note below
+#   - libbacktrace
+# The latter is a widely-used cross-platform open source library available at
+# https://github.com/ianlancetaylor/libbacktrace.  On some POSIX plaforms it's already either installed on the system
+# (eg see https://man7.org/linux/man-pages/man3/backtrace.3.html) or available as an optional component of the GCC
+# compiler.  However, it seems this is something that can change over time.  It's a small and stable library, so we
+# just build it from sources -- which is done in the `bt` script when you run `bt setup all`.
+#
+# Just to make things extra fun, in 2021, the GNU compilers did away with libdl and incorporated its functionality into
+# libc, per section 2.3 of release info at https://sourceware.org/glibc/wiki/Release/2.34.  This means, if we're using
+# the GNU tool chain and libc is v2.34 or newer, then we should NOT look for libdl, as we won't find it!  To find the
+# version of libc, you execute 'ldd --version', which gives multi-line output, of which the first line will be something
+# such as:
+#    ldd (Ubuntu GLIBC 2.35-0ubuntu3.1) 2.35
+# In this case, that means libc is version 2.35
+#
+# If we _don't_ need libdl, we just create an "unrequired" dependency.  This saves having to repeat a bunch of logic
+# later on when we get to the main build.
+#
+needLibdl = false
+if compiler.get_id() == 'gcc'
+   lddOutput = run_command('ldd', '--version', check: true).stdout().strip()
+   lddOutputLine1 = lddOutput.split('\n')[0]
+   message('ldd --version gives:', lddOutputLine1)
+   libcVersion = lddOutputLine1.split(' ')[-1]
+   message('libc version is:', libcVersion)
+   if libcVersion.version_compare('<2.34')
+      needLibdl = true
+   endif
+endif
+dlDependency = dependency('dl', required : needLibdl, static : true)
+#
+# On newer versions of Meson (0.62.0 and newer), the dependency has special-case code for dealing with libdl being built
+# into the compiler.  This means on Mac, for instance, we can have dlDependency.found() is true without there being any
```

Comment:
This is actually in large part why Meson has a dl dependency. You shouldn't need to mess with ldd -- because `dependency('dl')` will check for the dl functionality with two possible "dependency providers":
- check to see if `dlopen()` is available in libc
- check to see if an external `dl` library is available

But on versions of Meson before 0.62.0, it will simply not find anything at all, even if needLibdl is true. So Meson will just fail in that case.

Granted, if a case-sensitive cmake `Finddl.cmake` exists then meson will fall back on that for meson versions <0.62.0, but I am not aware of cmake itself shipping with such a thing.

If you need to support meson 0.60.0, then I'd suggest not bothering with ldd at all, and just using this:

```meson
if cc.has_function('dlopen', prefix: '#include <dlfcn.h>')
  dl_dep = declare_dependency()
else
  dl_dep = cc.find_library('dl', has_headers: 'dlfcn.h')
endif
```

Reply:
Ah, that's much more elegant.  Thanks!

Reply:
Except that it doesn't work on Windows. :crying_cat_face:
Well, we'll do the new way on Mac and Linux at least.

Reply:
What does dl do on Windows anyway? Is that just when building using a mingw-based compiler?

Reply:
Hmm, good question.  Maybe I'll try not having it as a dependency at all on Windows.